torch/_functorch/aot_autograd.py

```diff
@@ -918,6 +920,7 @@ def aot_export_module(
     # If trace_joint is True, we expect your module to return a scalar loss.
     # Your module can return multiple outputs, so you must specify which output the loss is.
     output_loss_index: Optional[int] = None,
+    pre_dispatch: bool = False,
```

Comment:
let's add some asserts: we should probably assert if someone passes in `trace_joint=True, pre_dispatch=True`, since we would never actually trace out a joint graph in this case (we are only able to trace out a joint graph by tracing through autograd to get out a backward graph, but pre_dispatch prevents us from tracing autograd)

Reply:
Can you also assert somewhere that we never had `AOTConfig.is_export == False and AOTConfig.pre_dispatch == True`? Right now we only promise to allow pre-dispatch in the export path